AK/HashTable.h

```diff
@@ -319,7 +326,20 @@ class HashTable {
 
     [[nodiscard]] ConstIterator find(T const& value) const
     {
-        return find(TraitsForT::hash(value), [&](auto& other) { return TraitsForT::equals(value, other); });
+        return find(TraitsForT::hash(value), [&](T const& other) { return TraitsForT::equals(value, other); });
+    }
+    // FIXME: Use some sort of Traits to get the comparison operation
+    // FIXME: Support for predicates, while guaranteeing that the predicate call
+    //        does not call a non trivial constructor each time invoked
+    template<typename K>
+    [[nodiscard]] Iterator find(K const& value) requires(!IsTriviallyConstructible<T, K> && requires { declval<K>() == declval<T>(); })
+    {
+        return find(Traits<K>::hash(value), [&](T const& other) { return value == other; });
+    }
+    template<typename K>
+    [[nodiscard]] ConstIterator find(K const& value) const requires(!IsTriviallyConstructible<T, K> && requires { declval<K>() == declval<T>(); })
```

Comment:
Since that requires clause is repeated so many times, please move it out to a concept or constexpr bool, as a bonus, the given name should make it clear what exactly these overloads are doing